core/src/main/java/org/elasticsearch/index/shard/DocsStats.java

```diff
@@ -57,16 +62,38 @@ public long getDeleted() {
         return this.deleted;
     }
 
+    /**
+     * Returns the total size in bytes of all documents in this stats.
+     * This value may be more reliable than {@link StoreStats#getSizeInBytes()} in estimating the index size.
+     */
+    public long getTotalSizeInBytes() {
+        return totalSizeInBytes;
+    }
+
+    /**
+     * Returns the average size in bytes of all documents in this stats.
+     */
+    public long getAverageSizeInBytes() {
+        long totalDocs = count + deleted;
+        return totalDocs == 0 ? 0 : totalSizeInBytes / totalDocs;
+    }
+
     @Override
     public void readFrom(StreamInput in) throws IOException {
         count = in.readVLong();
         deleted = in.readVLong();
+        if (in.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {
+            totalSizeInBytes = in.readVLong();
```

Comment:
Should we set `totalSizeInBytes` to -1 to indicate that it cannot be read and not that there are 0 bytes in use for 6.x nodes?

Reply:
Done https://github.com/elastic/elasticsearch/pull/27117/commits/96e9be768ed9c6a924dc1810803cb532b64dfb16